alacritty_terminal/src/grid/mod.rs

```diff
@@ -508,11 +558,33 @@ impl<T: GridCell + Copy + Clone> Grid<T> {
         }
     }
 
+    pub fn clear_viewport(&mut self, template: &T) {
+        // TODO: More efficiently using occ.
+        let mut iter = self.iter_from(Point { line: 0, col: Column(0) });
```

Comment:
Found the cause for the second issue!

```suggestion
        let mut iter = self.iter_from(Point { line: 0, col: self.num_cols() });
```